src/klein/resource.py

```diff
@@ -75,7 +75,10 @@ def _extractURLparts(request):
     @rtype: L{tuple} of L{unicode}, L{unicode}, L{int}, L{unicode}, L{unicode}
     """
     server_name = request.getRequestHostname()
-    server_port = request.getHost().port
+    if hasattr(request.getHost(), 'port'):
+        server_port = request.getHost().port
+    else:
+        server_port = 0
```

Comment:
In the lines just below this, `server_port` is appended to `server_name` unless it's one of the standard HTTP or HTTPS ports. This means that if we're listening on port 80 without TLS we end up with `example.com` as the `server_name`, but if we're listening on port 8080 we get `example.com:8080`.